src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            grow();
+        }
+        data[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index out of bound exception");
+        }
+        if (size == data.length) {
+            grow();
+        }
+        if (index < size) {
+            System.arraycopy(data, index, data, index + 1, size - index);
+        }
+        data[index] = value;
+        size += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't get element. Element does not exist for this index");
+        }
+        return data[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't set element. Element does not exist for this index");
+        }
+        data[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't remove element. Element does not exist for this index");
+        }
+        T tmp = data[index];
+        for (int i = index; i < size - 1; i++) {
+            data[i] = data[i + 1];
+        }
+        size -= 1;
+        return tmp;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size(); i++) {
+            if ((data[i] == null && element == null)
+                    || (data[i] != null && data[i].equals(element))) {
+                T tmp = data[i];
+                remove(i);
+                return tmp;
+            }
+        }
+        throw new NoSuchElementException();
```

Comment:
include element in the message

Reply:
not fixed, add description include element in the message